corehq/apps/accounting/models.py

```diff
@@ -1434,6 +1434,8 @@ def send_ending_reminder_email(self):
         Sends a reminder email to the emails specified in the accounting
         contacts that the subscription will end on the specified end date.
         """
+        if self.do_not_email:
```

Comment:
this will work although you could avoid making this function more complicated by putting the logic here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/accounting/tasks.py#L275

also, by filtering subscriptions at query time instead of later, you do less work